### PR TITLE
UI confirm flight implementaion

### LINKED ISSUE #158: Integrate Confirm Flight UI
Connect the Confirm flight screen to the navigation and handle the appropriate modelView updates

### PR DESCRIPTION
**Pull request by hacknstack**
Implemented the confirm flight UI with screen.
Added screen tests and added a "confirm" button.
Closes #158 
Please refer to PR : #177 for UI information.

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt [hunk 1]
```diff
@@ -41,7 +44,11 @@ import java.time.format.DateTimeFormatter
  * @param plannedFlight The planned flight for which the confirmation screen is displayed.
  */
 @Composable
-fun confirmationScreen(plannedFlight: PlannedFlight) {
+fun confirmationScreen(
+    plannedFlight: PlannedFlight,
+    navController: NavController,
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt:49]: do you use navController in this Composable? If yes, try to keep the navConstorller only in the toplevel Composable and work with onClick lambdas e.g.
if you use it for the "back" button define a onBackClick lambda

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt [hunk 2]
```diff
@@ -41,7 +44,11 @@ import java.time.format.DateTimeFormatter
  * @param plannedFlight The planned flight for which the confirmation screen is displayed.
  */
 @Composable
-fun confirmationScreen(plannedFlight: PlannedFlight) {
+fun confirmationScreen(
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt:47]: by convention so far we've used -Screen as a suffix only for the toplevel composables.

File: app/src/main/java/ch/epfl/skysync/screens/ConfirmFlightScreen.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import ch.epfl.skysync.models.flight.FlightType
+import ch.epfl.skysync.models.flight.PlannedFlight
+import ch.epfl.skysync.models.flight.Role
+import ch.epfl.skysync.models.flight.RoleType
+import ch.epfl.skysync.models.flight.Team
+import ch.epfl.skysync.models.flight.Vehicle
+import java.time.LocalDate
+
+@Composable
+fun confirmationScreenHardCoded(navController: NavController) {
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/screens/ConfirmFlightScreen.kt:19]: you can actually already (almost) fully connect your screen to the navigation. Check out how the ViewModel is init. for the`FlightDetailsScreen`. Your toplevel composable should take the arguments: 

```
navCtr: NavHostController,
flightId:String
vm: FlighstViewModel
```
Inside your toplevel function you ignore the flightId and viewModel for now and use the dummyFlight as is as well as the dummy onConfirmationClick

**Review by RAFInery — APPROVED** [review]

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt (hunk 2 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt]: fixed

File: app/src/main/java/ch/epfl/skysync/screens/ConfirmFlightScreen.kt (hunk 3 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/ConfirmFlightScreen.kt:20]: Done

File: app/src/main/java/ch/epfl/skysync/screens/flightDetail/FlightDetailsScreen.kt [hunk 4]
```diff
@@ -34,7 +34,7 @@ fun FlightDetailScreen(
         backClick = { navController.popBackStack() },
         deleteClick = { navController.popBackStack() },
         editClick = {},
-        confirmClick = {},
+        confirmClick = { navController.navigate("Confirm Flight") },
```
[the remark this answers is not in the document]
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/screens/flightDetail/FlightDetailsScreen.kt]: Fixed

File: app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt (hunk 1 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/ConfirmFlightUI.kt]: fixed

**Pull request merged**